textdb/textdb-exp/src/test/java/edu/uci/ics/textdb/exp/common/PredicateBaseTest.java

```diff
@@ -0,0 +1,192 @@
+package edu.uci.ics.textdb.exp.common;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.uci.ics.textdb.exp.dictionarymatcher.Dictionary;
+import edu.uci.ics.textdb.exp.dictionarymatcher.DictionaryPredicate;
+import edu.uci.ics.textdb.exp.dictionarymatcher.DictionarySourcePredicate;
+import edu.uci.ics.textdb.exp.fuzzytokenmatcher.FuzzyTokenPredicate;
+import edu.uci.ics.textdb.exp.fuzzytokenmatcher.FuzzyTokenSourcePredicate;
+import edu.uci.ics.textdb.exp.join.JoinDistancePredicate;
+import edu.uci.ics.textdb.exp.join.SimilarityJoinPredicate;
+import edu.uci.ics.textdb.exp.keywordmatcher.KeywordMatchingType;
+import edu.uci.ics.textdb.exp.keywordmatcher.KeywordPredicate;
+import edu.uci.ics.textdb.exp.keywordmatcher.KeywordSourcePredicate;
+import edu.uci.ics.textdb.exp.nlp.entity.NlpEntityPredicate;
+import edu.uci.ics.textdb.exp.nlp.entity.NlpEntityType;
+import edu.uci.ics.textdb.exp.nlp.sentiment.NlpSentimentPredicate;
+import edu.uci.ics.textdb.exp.projection.ProjectionPredicate;
+import edu.uci.ics.textdb.exp.regexmatcher.RegexPredicate;
+import edu.uci.ics.textdb.exp.regexmatcher.RegexSourcePredicate;
+import edu.uci.ics.textdb.exp.regexsplit.RegexSplitPredicate;
+import edu.uci.ics.textdb.exp.regexsplit.RegexSplitPredicate.SplitType;
+import edu.uci.ics.textdb.exp.sampler.SamplerPredicate;
+import edu.uci.ics.textdb.exp.sampler.SamplerPredicate.SampleType;
+import edu.uci.ics.textdb.exp.source.file.FileSourcePredicate;
+import edu.uci.ics.textdb.exp.source.scan.ScanSourcePredicate;
+import junit.framework.Assert;
+
+public class PredicateBaseTest {
+    
+    /*
+     * A helper test function to assert if a predicate is 
+     *   serialized / deserialized correctly
+     *   by converting predicate to json, then back to predicate, then back to json again,
+     *   assert two json strings are the same, (test if anything changed between conversions)
+     *   and the json contains "operatorType" and "id". (test if the predicate is registered in PredicateBase)
+     *   
+     */
+    public static void testPredicate(PredicateBase predicate) throws Exception {  
+        ObjectMapper objectMapper = new ObjectMapper();
+        String predicateJson = objectMapper.writeValueAsString(predicate);
+        PredicateBase resultPredicate = objectMapper.readValue(predicateJson, PredicateBase.class);
+        String resultPredicateJson = objectMapper.writeValueAsString(resultPredicate);
+        
+        System.out.println(predicateJson);
```

Comment:
for convenient debugging purposes, although I'd like to keep it, but I think removing it is better